nevergrad/parametrization/data.py

```
@@ -0,0 +1,353 @@
+            self.register_cheap_constraint(checker)
+        else:
+            raise ValueError(f"Unknown method {method}")
+        self.bounds = bounds  # type: ignore
+        self.full_range_sampling = full_range_sampling
+        # warn if sigma is too large for range
+        if both_bounds and method != "tanh":  # tanh goes to infinity anyway
+            std_bounds = tuple(self._to_std_space(b) for b in self.bounds)  # type: ignore
+            min_dist = np.min(np.abs(std_bounds[0] - std_bounds[1]).ravel())
+            if min_dist < 3.0:
+                warnings.warn(f"Bounds are {min_dist} sigma away from each other at the closest, "
+                              "you should aim for at least 3 for better quality.")
+        return self
+
+    def set_mutation(self: A, sigma: t.Optional[t.Union[float, "Array"]] = None, exponent: t.Optional[float] = None) -> A:
+        """Output will be cast to integer(s) through deterministic rounding.
+
+        Parameters
+        ----------
+        sigma: Array/Log or float
+            The standard deviation of the mutation. If a Parameter is provided, it will replace the current
+            value. If a float is provided, it will either replace a previous float value, or update the value
+            of the Parameter.
+        exponent: float
+            exponent for the logarithmic mode. With the default sigma=1, using exponent=2 will perform
+            x2 or /2 "on average" on the value at each mutation.
+
+        Returns
+        -------
+        self
+        """
+        if sigma is not None:
+            # just replace if an actual Parameter is provided as sigma, else update value (parametrized or not)
+            if isinstance(sigma, core.Parameter) or isinstance(self.subparameters._parameters["sigma"], core.Constant):
+                self.subparameters._parameters["sigma"] = core.as_parameter(sigma)
+            else:
+                self.sigma.value = sigma  # type: ignore
```

nit: Not sure if relevant here, but I would prefer to rather `assert isinstance(sigma, float)` if that can avoid a type ignore - and silent error fail

---

It can actually be a `np.ndarray` or a `float`, which makes typing a bit complicated :s